File: Distinct Difference.py
from typing import List

class Solution:
    def getDistinctDifference(self, N : int, A : List[int]) -> List[int]:
        left = [0 for i in range(N)]
        se = set()
        for i in range(1,N):
            se.add(A[i-1])
            left[i] = len(se)
        diff = [0 for i in range(N)]
        se = set()
        for i in range(N-2,-1,-1):
            se.add(A[i+1])
            diff[i] = left[i]-len(se)
        diff[-1] = left[-1]
        return diff

File: Distinct Difference.cpp
class Solution {
  public:
    vector<int> getDistinctDifference(int N, vector<int> &A) {
        // code here
        int i,j;
    	std::map<int, int> l,r;
    	for(i=0;i<N;i++) r[A[i]]++;
    	std::vector<int> v;
    	for(i=0;i<N;i++){
    		r[A[i]]--;
    		if(r[A[i]]==0) r.erase(A[i]);
    		v.push_back(l.size()-r.size());
    		l[A[i]]++;
    	}
    	return v;
    }
};
